src/test/java/org/sonarsource/sonarlint/ls/notebooks/VersionedOpenNotebookTests.java

```diff
@@ -494,4 +511,25 @@ public static VersionedOpenNotebook createTestNotebookWithThreeCells(URI tmpUri)
     return VersionedOpenNotebook.create(tmpUri, 1, List.of(cell1, cell2, cell3), mock(NotebookDiagnosticPublisher.class));
   }
 
+  private void manuallyReindexCellLines(VersionedOpenNotebook underTest) {
+    var fakeFindingDto = new RaisedIssueDto(
+      UUID.randomUUID(),
+      null,
+      "ruleKey",
+      "message",
+      IssueSeverity.BLOCKER,
+      RuleType.BUG,
+      CleanCodeAttribute.TRUSTWORTHY,
+      List.of(),
+      Instant.now(),
+      true,
+      false,
+      new TextRangeDto(1, 0, 1, 0),
+      List.of(),
+      List.of(),
+      null
+    );
+    underTest.toCellIssue(new DelegatingIssue(fakeFindingDto, URI.create("")));
```

Comment:
In reality this method will always be called before publishing the diagnostics. Here we fake the call.